libraries/chain/transaction_context.cpp

```diff
@@ -101,17 +106,21 @@ namespace eosio { namespace chain {
 
       net_limit = rl.get_block_net_limit();
 
-      objective_duration_limit = fc::microseconds( rl.get_block_cpu_limit() );
+      if( is_read_only() ) {
+         objective_duration_limit = read_only_trx_cpu_usage_limit;
```

Comment:
Not sure we should classify this as an objective limit. It certainly will not be once it is made configurable.
Probably clearer to add another section like:
```
      if( is_read_only() ) {
         // Possibly limit deadline to node configurable read_only_trx_cpu_usage_limit 
         if( start + read_only_trx_cpu_usage_limit ) <= _deadline ) {
            _deadline = start + read_only_trx_cpu_usage_limit ;
            billing_timer_exception_code = tx_cpu_usage_exceeded::code_value;
         }
```
Down below.
Will need to also report that differently in https://github.com/AntelopeIO/leap/pull/581

Reply:
I will add a new cpu exceeded type after #581 is merged.